keras_cv/models/object_detection/yolo_v8/yolo_v8_backbone.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import copy
+
+import tensorflow as tf
+from keras import layers
+from tensorflow import keras
+
+from keras_cv.models.backbones.backbone import Backbone
+from keras_cv.models.object_detection.yolo_v8.yolo_v8_backbone_presets import (
+    backbone_presets,
+)
+from keras_cv.models.object_detection.yolo_v8.yolo_v8_backbone_presets import (
+    backbone_presets_with_weights,
+)
+from keras_cv.models.object_detection.yolo_v8.yolo_v8_layers import (
+    apply_conv_bn,
+)
+from keras_cv.models.object_detection.yolo_v8.yolo_v8_layers import (
+    apply_csp_block,
+)
+from keras_cv.utils.python_utils import classproperty
+
+
+def apply_spatial_pyramid_pooling_fast(
+    inputs, pool_size=5, activation="swish", name=None
+):
+    channel_axis = -1
+    input_channels = inputs.shape[channel_axis]
+    hidden_channels = int(input_channels // 2)
+
+    nn = apply_conv_bn(
+        inputs,
+        hidden_channels,
+        kernel_size=1,
+        activation=activation,
+        name=f"{name}_pre",
+    )
+    pool_1 = layers.MaxPool2D(
+        pool_size=pool_size, strides=1, padding="same", name=f"{name}_pool1"
+    )(nn)
+    pool_2 = layers.MaxPool2D(
+        pool_size=pool_size, strides=1, padding="same", name=f"{name}_pool2"
+    )(pool_1)
+    pool_3 = layers.MaxPool2D(
+        pool_size=pool_size, strides=1, padding="same", name=f"{name}_pool3"
+    )(pool_2)
+
+    out = tf.concat([nn, pool_1, pool_2, pool_3], axis=channel_axis)
+    out = apply_conv_bn(
+        out,
+        input_channels,
+        kernel_size=1,
+        activation=activation,
+        name=f"{name}_output",
+    )
+    return out
+
+
+# TODO(ianstenbit) This should probably just be a CSPDarknet
+@keras.utils.register_keras_serializable(package="keras_cv.models")
```

Comment:
I'll avoid comments here assuming that this will be subbed out eventually

Reply:
Yeah, I didn't polish this backbone because it should be ephemeral.